beater/server_test.go

```diff
@@ -241,11 +241,11 @@ func TestServerSourcemapBadConfig(t *testing.T) {
 	ucfg, err := common.NewConfigFrom(m{"rum": m{"enabled": true, "source_mapping": m{"elasticsearch": m{"hosts": []string{}}}}})
 	require.NoError(t, err)
 	s, teardown, err := setupServer(t, ucfg, nil, nil)
-	require.Nil(t, s)
+	require.NotNil(t, s)
```

Comment:
Somewhere during development I found this error:
`./apm-server  -e -E apm-server.api_key.enabled=true -E apm-server.api_key.elasticsearch.ssl.enabled=false` 
```Exiting: Error processing configuration: missing required field accessing 'apm-server.api_key.elasticsearch.hosts' (source:'command line flag')
n```
It was fixed by removing the required flag here https://github.com/elastic/apm-server/blob/master/elasticsearch/config.go#L47, with this consequence. I am fine either way.

Reply:
Gotcha, fair enough. I think we need to update the test to use something else that's invalid, since it's meant to be testing invalid sourcemap config.

Reply:
In the other PR I've just skipped the test, reverting the change.